v5 release

Gauthamastro opened the pull request:
Features included

1. Orderbook v2
2. Thea protocol 
3. XCM Assets handlers
4. Crowdloan pallet
5. Updates substrate to polkadot-v0.9.37

serhii-temchenko committed:
style(formatting): Code style fixes

serhii-temchenko committed:
chore(ci): Bumped cache action version

serhii-temchenko committed:
temp(ci): Disable CI cache to test build

serhii-temchenko committed:
Revert "temp(ci): Disable CI cache to test build"

This reverts commit 758cd9f736ffde8f35f62618fc138b2cc15777c7.

serhii-temchenko committed:
fix(typo): Fixed typo

serhii-temchenko committed:
Merge branch 'Develop' into gj-parachain

Gauthamastro committed:
Changes related to Parachain and Thea (#741)

35359595 committed:
formatting and minor fixes

35359595 committed:
initial set id change test

35359595 committed:
flow updates

serhii-temchenko committed:
docs(changelog): Added changelog

Gauthamastro committed:
Update client RPC

Gauthamastro committed:
Fixed the RPC error

Gauthamastro committed:
Add logic for handling validator set id.

Gauthamastro committed:
Fix compilation

Gauthamastro committed:
Fix benchmarking.rs

Gauthamastro committed:
Add logic for handling validator set id in message handler

Gauthamastro committed:
Remove empty file

Gauthamastro committed:
Add governance extrinsic for updating last processed nonce

Gauthamastro committed:
Expiry thea message after a delay

Gauthamastro committed:
Add sanity check for foreign chain thea auth initialization.

Gauthamastro committed:
Fix compilation

Gauthamastro committed:
Merge remote-tracking branch 'origin/gj-thea-governance-extrinsic' into gj-thea-governance-extrinsic

# Conflicts:
#	clients/thea/src/worker.rs
#	primitives/thea/src/lib.rs

Gauthamastro committed:
Fix merge issues

Gauthamastro committed:
Another way to fetch auths

Gauthamastro committed:
Merge remote-tracking branch 'origin/gj-thea-governance-extrinsic' into gj-thea-governance-extrinsic

Gauthamastro committed:
Fix merge issues

serhii-temchenko committed:
Merge branch 'Develop' into iv-thea-tests

35359595 committed:
commented out thea tests for now

Gauthamastro committed:
Get CI passing

Gauthamastro committed:
Iv thea tests (#748)

## Describe your changes
Thea client testsing based on sp_network

## Issue ticket number and link
717 of Thea

## Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.
- [x] I removed all Clippy and Formatting Warnings. 
- [x] I added required Copyrights.

Gauthamastro committed:
Merge branch 'Develop' into gj-thea-validator-set-id

Gauthamastro committed:
Get CI passing

Gauthamastro committed:
Add Validator Set id grouping for Thea. (#749)

1. Adds super majority check for incoming messages
2. Handles incoming messages based on validator set id.

Gauthamastro committed:
Merge branch 'Develop' into gj-thea-governance-extrinsic

# Conflicts:
#	clients/thea/src/worker.rs
#	pallets/ocex/src/benchmarking.rs

Gauthamastro committed:
Get CI passing

Gauthamastro committed:
Add a governance extrinsic to update incoming nonce (#750)

1. Adds new governance extrinsic to skip nonces incase of message
delivery failures.
2. Adds an expiry for message cache in Thea client.

Gauthamastro committed:
Merge branch 'Develop' into changelog

Gauthamastro committed:
Added changelog (#747)

Added release notes (changelog) to track changes presented in a new
releases. Currently covers new future changes in an new release.

FYI @pavankanteti

Gauthamastro committed:
Remove seed from derivation in BLS

Gauthamastro committed:
Update lock file

Gauthamastro committed:
Add cli flag for thea foreign chain connector

Gauthamastro committed:
Add new extrinsic for parachain withdraw

Gauthamastro committed:
Add new extrinsic for parachain withdraw

Gauthamastro committed:
Add cli flag for thea foreign chain connector (#752)

Adds a cli flag for Thea foreign chain url

zktony committed:
Added asset creator

Gauthamastro committed:
fix tests

Gauthamastro committed:
Added asset creator (#753)

## Describe your changes
* If Minter fn doesn't find an asset, it will create a new one.

Gauthamastro committed:
Adds ValidatorSetId grouping for OB

Gauthamastro committed:
Adds ValidatorSetId based grouping for OB

Gauthamastro committed:
Store incoming messages on foreign chains

Gauthamastro committed:
Store incoming messages on foreign chains

Gauthamastro committed:
Add unique identifier for thea deposits and withdrawals

Gauthamastro committed:
Unique Identifier for Thea deposits and withdrawals (#757)

Gauthamastro committed:
Store Incoming Messages on Foreign chains (#756)

Gauthamastro committed:
Session Changes are grouped by Validator SetId in OCEX pallet (#755)

Updated pending snapshot implementation.

Gauthamastro committed:
Add id to WithdrawalQueued

Gauthamastro commented:
We will merge this after 200 node test and try runtime test. Don't merge before that.

Gauthamastro committed:
Fix PDX-110

Gauthamastro committed:
Update test.rs

Gauthamastro committed:
Fixes PDX-110 (#759)

Gauthamastro committed:
add panic messages for bls primitives

Gauthamastro committed:
Merge remote-tracking branch 'origin/Develop' into Develop

Gauthamastro committed:
cargo fmt

serhii-temchenko commented on .github/workflows/ci.yml:
```diff
@@ -73,11 +73,12 @@ jobs:
-          cargo clean
-          cargo build --release --features runtime-benchmarks
-          RUSTFLAGS="-D warnings" cargo build
+      - name: Build Project with runtime benchmarks
+        run: cargo build --features runtime-benchmarks
```
Test run of generating weights should be included as well.
Added in PR #760.

serhii-temchenko commented on .github/workflows/ci.yml:
```diff
@@ -100,6 +101,17 @@ jobs:
         with:
           name: code-coverage-report
           path: cobertura.xml
+#      - name: Zip compiled binaries
+#        if: contains(github.ref, 'Develop')
+#        # Zipping is required since billing is based on the raw uploaded size.
+#        run: zip -r -j polkadex-node.zip ./target/release/polkadex-node
+#      - name: Upload zipped binaries as an artifact
+#        if: contains(github.ref, 'Develop')
+#        uses: actions/upload-artifact@v3
+#        with:
+#          name: polkadex-node
+#          path: ./polkadex-node.zip
+#          if-no-files-found: error
```
@Gauthamastro can we uncomment this? its really helpful for the e2e testing things from other repos...

Gauthamastro replied:
Yes

serhii-temchenko commented on node/Cargo.toml:
```diff
@@ -18,107 +18,100 @@ structopt = { version = "0.3.8", optional = true }
+sc-consensus = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.37" }
+sc-consensus-babe =  { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.37" }
+sc-consensus-epochs =  { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.37" }
+sp-tracing =  { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.37" }
```
Code style.
Fixed in PR https://github.com/Polkadex-Substrate/Polkadex/pull/760.

serhii-temchenko replied:
Fixed in pr #760

serhii-temchenko commented on pallets/chainbridge/src/lib.rs:
```diff
@@ -349,8 +359,10 @@ pub mod pallet {
-        (dispatch_info.weight.saturating_add(195_000_000), dispatch_info.class, Pays::Yes)
+        (dispatch_info.weight, dispatch_info.class, Pays::Yes)
         })]
+		// TODO: benchmark this function too and add to proposal weight above
```
Unresolved todo

Gauthamastro replied:
that's fine, chainbridge will be removed in next to next release, the current weight will not cause big issues.

serhii-temchenko commented on pallets/chainbridge/src/lib.rs:
```diff
@@ -397,16 +410,17 @@ pub mod pallet {
-        (dispatch_info.weight.saturating_add(195_000_000), dispatch_info.class, Pays::Yes)
+        (dispatch_info.weight, dispatch_info.class, Pays::Yes)
         })]
+		// TODO: Benchmark this function too.
```
Unresolved todo

Gauthamastro replied:
Same as above

serhii-temchenko committed:
chore(ci): Added weights generation test action to the CI

serhii-temchenko commented on CHANGELOG.md:
```diff
@@ -0,0 +1,234 @@
+
+## [Unreleased]
+
+## [5.0.0] - 2023-05-16
```
Update date when everything will be ready for the release

serhii-temchenko replied:
Updated in PR #763

serhii-temchenko commented on CHANGELOG.md:
Extend with recent PR's merges... @serhii-temchenko

serhii-temchenko replied:
Updated in PR #763

serhii-temchenko committed:
chore(ci): Added .toml files linter and check in the CI

serhii-temchenko committed:
style(formatter): Added taplo formatter configuration

serhii-temchenko committed:
style(toml): Fixed formatting issues in all .toml files

serhii-temchenko commented on clients/orderbook/rpc/src/lib.rs:
No proper documentation. Need to create an issue for this.

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko committed:
chore(ci): Fixed linter setup for the CI

serhii-temchenko commented on clients/orderbook/src/error.rs:
```diff
@@ -0,0 +1,100 @@
+	ProxyAccountNotFound,
+	#[error("Proxy not associated with main")]
+	ProxyNotAssociatedWithMain,
+	// #[error("Error while snapshot signing")]
+	// SnapshotSigningFailed,
```
Could be removed?

serhii-temchenko replied:
Removed in pr #760

serhii-temchenko commented on clients/orderbook/src/gossip.rs:
No module level docs and not covered all public API properly.

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on clients/orderbook/src/keystore.rs:
No module level docs and not covered all public API properly.

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on clients/orderbook/src/lib.rs:
```diff
@@ -0,0 +1,200 @@
+	pub is_validator: bool,
+	/// Submit message link
+	pub message_sender_link: UnboundedReceiver<ObMessage>,
+	// Links between the block importer, the background voter and the RPC layer.
+	// pub links: BeefyVoterLinks<B>,
```
Could be removed?

serhii-temchenko replied:
Removed in pr #760

serhii-temchenko commented on clients/orderbook/src/lib.rs:
No module level docs.

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on clients/orderbook/src/tests/rpc.rs:
```diff
@@ -0,0 +1,192 @@
+		_group: Option<&'static str>,
+		_future: BoxFuture<'static, ()>,
+	) {
+		todo!()
```
What is about this todo?

Gauthamastro replied:
its not necessary for that test. changing to unreachable

serhii-temchenko commented on clients/orderbook/src/tests/rpc.rs:
```diff
@@ -0,0 +1,192 @@
+		_group: Option<&'static str>,
+		_future: BoxFuture<'static, ()>,
+	) {
+		todo!()
```
What is about this todo?

Gauthamastro replied:
Same

serhii-temchenko commented on clients/orderbook/src/tests/sync.rs:
```diff
@@ -0,0 +1,143 @@
+	// Let the network activity settle down.
+	testnet.run_until_idle().await;
+
+	// TODO: Fix this in the next release.
+	// The fullnodes are not recieving gossip in unit tests.
+	// let working_root = working_state_root.read();
+	// // Assert if the fullnode's working state root is updated.
+	// assert_eq!(sp_core::H256::from_slice(&*working_root),
+	// runtime.get_latest_snapshot().state_root)
```
Is "next release" is this release?

Gauthamastro replied:
No, it will be in v5.1.0

serhii-temchenko commented on clients/orderbook/src/utils.rs:
No documentation

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on clients/orderbook/src/worker.rs:
```diff
@@ -0,0 +1,1497 @@
+						}
+					}
+				} else {
+					// TODO: Reduce reputation if else block is happens
```
Still remains?

Gauthamastro replied:
v5.1.0

serhii-temchenko commented on clients/orderbook/src/worker.rs:
Lack of documentation on most of public API (and no module level docs)

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on clients/thea/src/connector/parachain.rs:
```diff
@@ -0,0 +1,130 @@
+
+		// Get validator set
+		let storage_address = parachain::storage().thea_message_handler().authorities(set_id);
+		// TODO: Get last finalized block hash
```
Still remaining?

Gauthamastro replied:
Resolved by #778

serhii-temchenko commented on clients/thea/src/connector/parachain.rs:
```diff
@@ -0,0 +1,130 @@
+	async fn check_thea_authority_initialization(&self) -> Result<bool, Error> {
+		// Get current validator set id
+		let storage_address = parachain::storage().thea_message_handler().validator_set_id();
+		// TODO: Get last finalized block hash
```
Still remaining?

serhii-temchenko commented on clients/thea/src/connector/parachain.rs:
```diff
@@ -0,0 +1,130 @@
+	async fn last_processed_nonce_from_native(&self) -> Result<u64, Error> {
+		// Read native network nonce from foreign chain
+		let storage_address = parachain::storage().thea_message_handler().incoming_nonce();
+		// TODO: Get last finalized block hash
```
Still remaining?

serhii-temchenko commented on clients/thea/src/connector/parachain.rs:
```diff
@@ -0,0 +1,130 @@
+		// Read thea messages from foreign chain
+		let storage_address =
+			parachain::storage().thea_message_handler().outgoing_messages(message.nonce);
+		// TODO: Get last finalized block hash
```
Still remaining?

serhii-temchenko commented on clients/thea/src/connector/parachain.rs:
```diff
@@ -0,0 +1,130 @@
+	async fn read_events(&self, nonce: u64) -> Result<Option<Message>, Error> {
+		// Read thea messages from foreign chain
+		let storage_address = parachain::storage().thea_message_handler().outgoing_messages(nonce);
+		// TODO: Get last finalized block hash
```
Still remaining?

serhii-temchenko commented on clients/thea/src/connector/parachain.rs:
No proper docs

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on clients/thea/src/connector/traits.rs:
Module level docs

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on clients/thea/src/connector/traits.rs:
```diff
@@ -0,0 +1,61 @@
+	async fn last_processed_nonce_from_native(&self) -> Result<u64, Error>;
+	/// Check if the foreign chain is initialized with thea validators
+	async fn check_thea_authority_initialization(&self) -> Result<bool, Error>;
+}
```
Proper documentation

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on clients/thea/src/gossip.rs:
Documentation

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on clients/thea/src/keystore.rs:
Documentation

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on clients/thea/src/lib.rs:
Crate level docs an public api

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on clients/thea/src/tests/mod.rs:
```diff
@@ -0,0 +1,440 @@
+};
+use tokio::time::Instant;
+
+//pub mod deposit;
+mod grandpa;
+//mod protocol;
+//pub mod withdrawal;
```
Commented out tests should be fixed and included back

Gauthamastro replied:
All disabled unit tests can be done in the next release.  Tracking issues: https://github.com/Polkadex-Substrate/Polkadex/issues/718
 https://github.com/Polkadex-Substrate/Polkadex/issues/717

serhii-temchenko commented on clients/thea/src/tests/protocol.rs:
```diff
@@ -0,0 +1,214 @@
+	testnet.peer(0).push_blocks(1, false);
+	testnet.run_until_sync().await;
+
+	// rotate id
+
+	// send foreign message with old id
+
+	// verify local parsing still works
+
+	// verify key and id updated on foreign chain
```
Required test logic is not implemented?

Gauthamastro replied:
Same as above, the issue is with the substrate libraries, so we are waiting to resolve that before enabling this.

serhii-temchenko commented on clients/thea/src/metrics.rs:
```diff
@@ -0,0 +1,91 @@
+}
+
+impl Metrics {
+	pub fn register(registry: &Registry) -> Result<Self, PrometheusError> {
```
Docs

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on clients/thea/src/metrics.rs:
```diff
@@ -0,0 +1,91 @@
+// Note: we use the `format` macro to convert an expr into a `u64`. This will fail,
+// if expr does not derive `Display`.
+#[macro_export]
+macro_rules! metric_set {
+	($self:ident, $m:ident, $v:expr) => {{
+		let val: u64 = format!("{}", $v).parse().unwrap();
+
+		if let Some(metrics) = $self.metrics.as_ref() {
+			metrics.$m.set(val);
+		}
+	}};
+}
+
+#[macro_export]
+macro_rules! metric_inc {
+	($self:ident, $m:ident) => {{
+		if let Some(metrics) = $self.metrics.as_ref() {
+			metrics.$m.inc();
+		}
+	}};
+}
+
+#[cfg(test)]
+#[macro_export]
+macro_rules! metric_get {
+	($self:ident, $m:ident) => {{
+		$self.metrics.as_ref().map(|metrics| metrics.$m.clone())
+	}};
+}
+
+#[macro_export]
+macro_rules! metric_add {
+	($self:ident, $m:ident, $v:expr) => {{
+		let val: u64 = format!("{}", $v).parse().unwrap();
+
+		if let Some(metrics) = $self.metrics.as_ref() {
+			metrics.$m.add(val);
+		}
+	}};
+}
```
Docs

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on clients/thea/src/types.rs:
Docs

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on clients/thea/src/worker.rs:
Module level and pub api docs

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on misc/load-testing/src/main.rs:
What this about? should be removed?

serhii-temchenko commented on node/src/benchmarking.rs:
```diff
@@ -19,7 +19,7 @@
 //!
 //! Should only be used for benchmarking as it may break in other contexts.
-
+#![allow(dead_code, warnings)]
```
I think this should not be here

serhii-temchenko replied:
For some reason members of this module are used in the `command` module, but warnings are still thrown. Leaving as it is...

serhii-temchenko commented on node/src/main.rs:
```diff
@@ -1,13 +1,17 @@
 mod rpc;
 
+// TODO: Remove this when libp2p enforces correct snow version
+use snow as _;
```
Is it still remains?

serhii-temchenko commented on pallets/asset-handler/src/lib.rs:
```diff
@@ -550,6 +597,70 @@ pub mod pallet {
 				.collect()
 		}
 
+		pub fn mint_thea_asset(
+			asset_id: u128,
+			recipient: T::AccountId,
+			amount: u128,
+		) -> Result<(), DispatchError> {
```
Docs

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on pallets/asset-handler/src/lib.rs:
Crate level docs

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on pallets/asset-handler/src/lib.rs:
```diff
@@ -550,6 +597,70 @@ pub mod pallet {
+			)
+		}
+
+		pub fn burn_thea_asset(
+			asset_id: u128,
+			who: T::AccountId,
+			amount: u128,
+		) -> Result<(), DispatchError> {
+			ensure!(amount > 0, Error::<T>::AmountCannotBeZero);
+			T::AssetManager::burn_from(asset_id, &who, amount)?;
+			Ok(())
+		}
+
+		pub fn get_asset_id(derived_asset_id: Vec<u8>) -> u128 {
+			let derived_asset_id_hash = &keccak_256(derived_asset_id.as_ref())[0..16];
+			let mut temp = [0u8; 16];
+			temp.copy_from_slice(derived_asset_id_hash);
+			u128::from_le_bytes(temp)
+		}
```
Docs

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on pallets/asset-handler/src/lib.rs:
```diff
@@ -567,4 +678,174 @@ pub mod pallet {
+			} else if T::Currency::free_balance(who) >= amount.saturated_into() {
+				WithdrawConsequence::Success
+			} else {
+				// TODO: Need a better error mapping
+				WithdrawConsequence::UnknownAsset
```
Still remaining?

Gauthamastro replied:
RuntimeTODOs will be done in the next release

serhii-temchenko commented on pallets/chainbridge/src/lib.rs:
```diff
@@ -257,7 +261,7 @@ pub mod pallet {
 			// Clear all bridge transfer data
 			BridgeEvents::<T>::kill();
-			0
+			Weight::default() // TODO: This is not zero
```
Still remaining?

Gauthamastro replied:
RuntimeTODOs will be done in the next release

serhii-temchenko commented on pallets/ocex/src/lib.rs:
Crate level and public api docs

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on pallets/ocex/src/tests.rs:
```diff
@@ -1323,275 +1362,267 @@ fn test_collect_fees_bad_origin() {
 
-#[test]
-fn test_allowlist_enclave_bad_origin() {
+// P.S. This was to apply a DDOS attack and see the response in the mock environment
+/* #[test]
+fn collect_fees_ddos(){
 	let account_id = create_account_id();
+	new_test_ext().execute_with(||{
+		// TODO! Discuss if this is expected behaviour, if not then could this be a potential DDOS?
+		for x in 0..10000000 {
+			assert_ok!(
+				OCEX::collect_fees(
+					RuntimeOrigin::signed(account_id.clone().into()),
+					x,
+					account_id.clone().into()
+				)
+			);
+		}
+	});
+} */
```
Should be uncommented and excluded from normal execution process with proper documentation?

Gauthamastro replied:
RuntimeTODOs will be done in the next release

serhii-temchenko commented on pallets/pdex-migration/src/lib.rs:
Crate level and public api docs

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on pallets/rewards/rpc/runtime-api/src/lib.rs:
Crate level docs

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on pallets/rewards/rpc/src/lib.rs:
Crate level docs

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on pallets/rewards/src/crowdloan_rewardees.rs:
Module level docs

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on pallets/rewards/src/lib.rs:
Crate level docs

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on pallets/support/src/lib.rs:
Crate level docs

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on pallets/thea-executor/src/lib.rs:
Crate level docs

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on pallets/thea-message-handler/src/lib.rs:
Crate level docs

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on pallets/thea/src/lib.rs:
Crate level docs

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko committed:
fix(benchmarks): Corrected fixture after related structs refactoring

serhii-temchenko committed:
fix(benchmarks): Removed wrongly added unwrap

serhii-temchenko committed:
style(dead-code): Removed commented out code parts

serhii-temchenko commented on primitives/thea/src/types.rs:
Docs

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on primitives/thea/src/test.rs:
```diff
@@ -0,0 +1,14 @@
+use sp_core::Pair;
+use sp_runtime::traits::IdentifyAccount;
+
+#[test]
+fn get_thea_public_key() {
+	let seed = "owner word vocal dose decline sunset battle example forget excite gentle waste//";
+	let idx = 3;
+	let thea = crate::crypto::Pair::from_string(
+		&(seed.to_owned() + idx.to_string().as_str() + "//thea"),
+		None,
+	)
+	.unwrap();
+	println!("public_key : {:?}", hex::encode(thea.public().into_account()));
+}
```
Doesn't looks like this test brings some value, I assume only for local/manual testing purpose. I think this should be removed?

Gauthamastro replied:
yes, we can remove this.

serhii-temchenko replied:
Removed in PR #768. Resolving this comment.

serhii-temchenko commented on primitives/thea/src/lib.rs:
Crate level and some public API (constants) docs are not present.

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on primitives/polkadex/src/withdrawal.rs:
Docs

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on primitives/polkadex/src/utils.rs:
Docs

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on primitives/polkadex/src/ocex.rs:
Docs

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on primitives/polkadex/src/lib.rs:
Pub API docs

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on primitives/polkadex/src/ingress.rs:
Docs

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on primitives/polkadex/src/fees.rs:
Docs

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on pallets/thea/src/session.rs:
```diff
@@ -0,0 +1,57 @@
+	where
+		I: Iterator<Item = (&'a T::AccountId, T::TheaId)>,
+	{
+		// TODO: what happens if someone changes their session key
+		// and queued_validators != validators in next set.???
```
Still remains?

Gauthamastro replied:
Pallet TODOs will be done in the next release

serhii-temchenko commented on primitives/bls-primitives/src/lib.rs:
Docs

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on primitives/orderbook/src/types.rs:
Docs

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on primitives/orderbook/src/lib.rs:
Crate level docs

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

serhii-temchenko commented on primitives/orderbook/src/constants.rs:
Docs on pub API

serhii-temchenko replied:
Should be resolved in the scope of separate issue #762. Resolving this comment for now.

Gauthamastro committed:
Release preparation improvements (#760)

In this PR:
- Extended CI with test of weights generation run;
- Added `.toml` files linter/formatter configuration (for
[taplo](https://github.com/tamasfe/taplo) linter) and CI step to check
and get rid of problem with `.toml` files formatting;
- Fixed formatting in all `.toml` files;

Gauthamastro committed:
Add logs to trade verification

Fix orderbook rebroadcast check

Gauthamastro committed:
Remove message from cache on expiry

Gauthamastro committed:
Avoid importing genesis block

Gauthamastro committed:
Avoid importing genesis block

35359595 commented on clients/orderbook/rpc/src/lib.rs:
```diff
@@ -0,0 +1,234 @@
+// Provides RPC methods for interacting with Orderbook.
+#[rpc(client, server)]
+pub trait OrderbookApi {
+	/// Returns hash of the latest Orderbook finalized block as seen by this client.
```
Not sure if this comment applies. Looks like it returns nothing.

35359595 commented on clients/orderbook/rpc/src/lib.rs:
```diff
@@ -0,0 +1,234 @@
+/// Implements the OrderbookApi RPC trait for interacting with Orderbook.
+pub struct OrderbookRpc<Runtime, Block, Client> {
+	tx: UnboundedSender<ObMessage>,
+	_executor: SubscriptionTaskExecutor,
```
this never used - can be removed?

35359595 commented on clients/orderbook/src/gossip.rs:
```diff
@@ -0,0 +1,229 @@
+				let latest_worker_nonce = *self.latest_worker_nonce.read();
+				if msg.worker_nonce > latest_worker_nonce {
+					// It's a new message so we process it and keep it in our pool
+					ValidationResult::ProcessAndKeep(topic::<B>())
```
isn't using `self.topic` cheaper than this [non constant] call?

Gauthamastro replied:
done

35359595 commented on clients/orderbook/src/gossip.rs:
```diff
@@ -0,0 +1,229 @@
+				}
+				// Validators only process it if the request is for nonces after
+				if *from >= self.last_snapshot.read().worker_nonce {
+					ValidationResult::ProcessAndKeep(topic::<B>())
```
isn't using `self.topic` cheaper than this [non constant] call?

35359595 commented on clients/orderbook/src/gossip.rs:
```diff
@@ -0,0 +1,229 @@
+				// We only process the request for last snapshot
+				if self.last_snapshot.read().snapshot_id == *snapshot_id {
+					self.message_cache.write().insert((msg_hash, peerid), Instant::now());
+					ValidationResult::ProcessAndKeep(topic::<B>())
```
isn't using `self.topic` cheaper than this [non constant] call?

35359595 commented on clients/orderbook/src/gossip.rs:
```diff
@@ -0,0 +1,229 @@
+					ValidationResult::Discard
+				} else {
+					self.message_cache.write().insert((msg_hash, peerid), Instant::now());
+					ValidationResult::ProcessAndDiscard(topic::<B>())
```
isn't using `self.topic` cheaper than this [non constant] call?

35359595 commented on clients/orderbook/src/gossip.rs:
```diff
@@ -0,0 +1,229 @@
+			GossipMessage::Want(_, _) => WANT_REBROADCAST_INTERVAL,
+			_ => REBROADCAST_INTERVAL,
+		};
+		if self.message_expired_check(message) {
```
can this be first check in this method? requires no allocation of `hash` and `interval` if false...

Gauthamastro replied:
resolved

serhii-temchenko reviewed (changes requested):
Weights for the pallets should be regenerated since they are outdated.
FYI @Gauthamastro, @zktony, @35359595

Gauthamastro committed:
remove updating trading pairs from block import

Gauthamastro committed:
Add dummy mode for thea

Gauthamastro committed:
add log for session change

Gauthamastro committed:
Fix comments

Gauthamastro committed:
Add a check for BLS keys

Gauthamastro committed:
Add some logs

Gauthamastro committed:
cargo fmt

serhii-temchenko committed:
docs(changelog): Updated changelog with recently merged PRs

Gauthamastro committed:
Add thea keys

Gauthamastro committed:
Add crowdloan verifier

Gauthamastro committed:
Add a README file

Gauthamastro committed:
Remove reward list

Gauthamastro committed:
clippy

serhii-temchenko committed:
Updated changelog (#763)

After recent merges to the `Develop` branch - changelog should be
updated with new updates.
In this PR added new changes to the changelog before release rolling
out.

Commit:
use of Percent instead of sat arithmetics

35359595 commented on clients/thea/src/gossip.rs:
```diff
@@ -0,0 +1,147 @@
+		if let Ok(thea_gossip_msg) = GossipMessage::decode(&mut data) {
+			// Check if we processed this message
+			if self.validate_message(&thea_gossip_msg) {
+				trace!(target:"thea-gossip", "Validation successfully for message: {thea_gossip_msg:?}");
```
we need to remove it from cache here?

Gauthamastro replied:
It's removed from Cache during rebroadcast check

35359595 commented on clients/thea/src/worker.rs:
```diff
@@ -0,0 +1,576 @@
+	_network: Arc<N>,
+	keystore: TheaKeyStore,
+	gossip_engine: GossipEngine<B>,
+	_gossip_validator: Arc<GossipValidator>,
```
we can remove this field, no need to hold it's ref here?

35359595 commented on clients/thea/src/worker.rs:
```diff
@@ -0,0 +1,576 @@
+	sync_oracle: SO,
+	metrics: Option<Metrics>,
+	is_validator: bool,
+	_network: Arc<N>,
```
we can remove this field, no need to hold it's ref here?

35359595 commented on clients/thea/src/worker.rs:
```diff
@@ -0,0 +1,576 @@
+	pub(crate) client: Arc<C>,
+	pub(crate) thea_network: Option<Network>,
+	// Payload to gossip message mapping
+	_backend: Arc<BE>,
```
we can remove this field, no need to hold it's ref here?

35359595 commented on clients/thea/src/worker.rs:
```diff
@@ -0,0 +1,576 @@
+	}
+
+	pub async fn check_message(&mut self, message: &GossipMessage) -> Result<bool, Error> {
+		// TODO: Do signature check here.
```
should we? :)

Gauthamastro replied:
We don't need that; libp2p already does that for us.

35359595 commented on clients/thea/src/worker.rs:
```diff
@@ -0,0 +1,576 @@
+	pub async fn process_gossip_message(
+		&mut self,
+		incoming_message: &mut GossipMessage,
+		_: Option<PeerId>,
```
do we need this param?

35359595 commented on clients/thea/src/worker.rs:
```diff
@@ -0,0 +1,576 @@
+							incoming_message.payload.clone(),
+							(Instant::now(), incoming_message.clone()),
+						);
+						// TODO: Send it back to network.
```
should we?

Gauthamastro replied:
It's outdated

35359595 reviewed (changes requested):
A lot of files missing copyrights and license information

Gauthamastro committed:
Add metadata for assetid

nuel77 committed:
add prefix for thea logs 🌉

nuel77 committed:
add prefix for orderbook logs 📒

nuel77 committed:
missed few

nuel77 committed:
fix fmt

Gauthamastro committed:
Made snapshot serialization deterministic

Gauthamastro committed:
Made snapshot serialization deterministic

Gauthamastro committed:
seperate orderbook and thea logs using prefix (#766)

## Describe your changes
Currently both orderbook and thea which share the same functionality has
similar logs. Added different prefix for each of them so its easier to
distinguish between them

Gauthamastro committed:
use of Percent instead of sat arithmetics (#765)

## Describe your changes
Audit 109 issue fix. (and same stuff in primitives)

## Issue ticket number and link
Thea 197

## Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.
- [x] I removed all Clippy and Formatting Warnings. 
- [x] I added required Copyrights.

Gauthamastro committed:
Merge branch 'Develop' into gj-fix-rebroadcast-check

# Conflicts:
#	clients/orderbook/src/worker.rs
#	clients/thea/src/lib.rs
#	primitives/thea/src/types.rs

nuel77 committed:
fix log

Gauthamastro committed:
Added logs for OCEX submit snapshot

serhii-temchenko committed:
chore(test): Removed useless test

Gauthamastro committed:
Taplo fmt

serhii-temchenko committed:
chore(cleanup): Removed default comments

nuel77 committed:
fix log

Gauthamastro committed:
Added logs for OCEX submit snapshot

Gauthamastro committed:
Merge remote-tracking branch 'origin/gj-fix-rebroadcast-check' into gj-fix-rebroadcast-check

Gauthamastro committed:
add unit for memory db serialization and deserialization

Gauthamastro committed:
Add new rpc for reading from storage

Gauthamastro committed:
Fixed the mutex bug

Gauthamastro committed:
typo

Gauthamastro committed:
Fix testss

serhii-temchenko committed:
feat(docs): Added copyrights and license documentation

Gauthamastro committed:
cargo fmt

Gauthamastro committed:
Enable validator for Want requests

Gauthamastro committed:
Fix RPC

Gauthamastro committed:
Add log for memory db keys

Gauthamastro committed:
revert RPC fix

Gauthamastro committed:
Fix comments

Gauthamastro committed:
Update working state root too.

Gauthamastro committed:
Add more logs

Gauthamastro committed:
Check if the main account exists in the trie

Gauthamastro committed:
Check if the main account exists in the trie

Gauthamastro committed:
Print trie empty()

Gauthamastro committed:
Add more logs

serhii-temchenko commented:
> A lot of files missing copyrights and license information

Resolved in the PR #770

Gauthamastro committed:
Get state from storage now gets summary from runtime

Gauthamastro committed:
Add more logs

Gauthamastro committed:
Add more logs and error

Gauthamastro committed:
Add more logs and error

Gauthamastro committed:
gossip message for sync

Gauthamastro committed:
logs

Gauthamastro committed:
Fix gossip validator

Gauthamastro committed:
Fix gossip validator

Gauthamastro committed:
more and more logs

Gauthamastro committed:
Fix finalization number

Gauthamastro committed:
Sync state if we are don't have latest state

Gauthamastro committed:
Sync state if we are don't have latest state

Gauthamastro committed:
Sync state if we are don't have latest state

Gauthamastro committed:
Sync state if we are don't have latest state

Gauthamastro committed:
Take finalized block from client

Gauthamastro committed:
Copyrights (#770)

## Describe your changes
Added copyrights in each `.rs` file which didn't had it before, docker
files, configs, shell scripts and root Cargo.toml files.
## Issue ticket number and link
#769 
## Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.
- [x] I removed all Clippy and Formatting Warnings. 
- [x] I added required Copyrights.

Gauthamastro committed:
Cleanup (#768)

In this PR removed useless test which gives no value but rather adds
code which should be maintained.
Additionally removed default comments in `cargo.toml` files.

Gauthamastro committed:
Add back trading pair

Gauthamastro committed:
Merge branch 'Develop' into gj-fix-rebroadcast-check

# Conflicts:
#	clients/orderbook/src/snapshot.rs
#	scripts/start_validator1.sh
#	scripts/start_validator2.sh
#	scripts/start_validator3.sh

Gauthamastro committed:
Clippy fixes

Gauthamastro committed:
Fix test for snapshot

Gauthamastro committed:
cargo fmt

Gauthamastro committed:
Fix toml files

Gauthamastro committed:
add one more log

Gauthamastro committed:
add one more log

Gauthamastro committed:
Add crowdloan address verifier tool (#764)

A tool to verify the excel sheet of contributor list with the hardcoded
map on the rewards pallet

Gauthamastro committed:
Add stid and worker nonce to withdrawals

Gauthamastro committed:
Add stid and worker nonce to withdrawals

Gauthamastro committed:
update events with correct decimal config

Gauthamastro committed:
Remove log

Gauthamastro committed:
Wait for block finality before proceeding

Gauthamastro committed:
Wait for block finality before proceeding

Gauthamastro committed:
Import blocks in order

Gauthamastro committed:
Add reset in user actions

Gauthamastro committed:
Add state version to snapshot summary

Gauthamastro committed:
Add state verison

Gauthamastro committed:
Add State version (#771)

## Describe your changes

The state version allows segregating messages into pre and post-recovery
of the engine, it avoid the orderbook client from executing old messages
after recovery

Gauthamastro committed:
Merge branch 'Develop' into gj-fix-rebroadcast-check

# Conflicts:
#	clients/orderbook/src/worker.rs

Gauthamastro committed:
Add reset in user actions

Gauthamastro committed:
Merge remote-tracking branch 'origin/Develop' into Develop

Gauthamastro committed:
Fix path in set-keys.sh

Gauthamastro committed:
Discard gossip messages from old state versions

Gauthamastro committed:
Discard gossip messages from old state versions

Gauthamastro committed:
Merge branch 'Develop' into gj-fix-rebroadcast-check

Gauthamastro committed:
Add state version in ObRecoveryState

Gauthamastro committed:
Add state version in ObRecoveryState (#772)

Gauthamastro committed:
Merge branch 'Develop' into gj-fix-rebroadcast-check

Gauthamastro committed:
Add state version in ObRecoveryState

Gauthamastro committed:
gossip reset to others

Gauthamastro committed:
Update worker nonce

Gauthamastro committed:
Fix gossip check

Gauthamastro committed:
Fix panic

Gauthamastro committed:
Update fixtures

Gauthamastro committed:
Update fixtures

Gauthamastro committed:
Merge branch 'Develop' into gj-fix-rebroadcast-check

Gauthamastro committed:
fix tests

Gauthamastro committed:
Gossip test

Gauthamastro committed:
Final list update

Gauthamastro committed:
Change governance origin for rewards

Gauthamastro committed:
Clippy

Gauthamastro committed:
Update Contributor list (#773)

Gauthamastro committed:
Merge branch 'Develop' into gj-fix-rebroadcast-check

# Conflicts:
#	Cargo.lock

Gauthamastro committed:
Add try runtime hooks

Gauthamastro committed:
Revert "Add try runtime hooks"

This reverts commit 4cb39cd42774a185e39bcb03e96779a5d385ae9a.

Gauthamastro committed:
Add try -runtime

Gauthamastro committed:
Add bls host function

Gauthamastro committed:
Merge remote-tracking branch 'origin/gj-fix-rebroadcast-check' into gj-fix-rebroadcast-check

Gauthamastro committed:
Add host function

zktony committed:
Fixed try-runtime

zktony committed:
fixed formatting

Gauthamastro committed:
Try Runtime (#774)

Gauthamastro committed:
Merge branch 'Develop' into gj-fix-rebroadcast-check

# Conflicts:
#	runtime/Cargo.toml
#	runtime/src/lib.rs

Gauthamastro committed:
Add host function

serhii-temchenko committed:
chore(docs): Updated docs

Gauthamastro committed:
1. Snapshot based on imported blks from engine
2. Pending snapshot check based on runtime for snapshot generation check

Gauthamastro committed:
Add default members and fix other issues

Gauthamastro committed:
Fix a bunch of things

zktony committed:
Added feature

Gauthamastro committed:
It's working

Gauthamastro committed:
Merge remote-tracking branch 'origin/gj-fix-rebroadcast-check' into gj-fix-rebroadcast-check

Gauthamastro committed:
It's working

Gauthamastro committed:
Correct versioning

Gauthamastro committed:
Correct versioning

Gauthamastro committed:
Fix try runtime

serhii-temchenko committed:
docs(762): Extended documentation for the primitives

serhii-temchenko committed:
docs(762): Extended documentation for the pallets

serhii-temchenko committed:
docs(762): Removed redundand todo

Gauthamastro committed:
All good

Gauthamastro committed:
Last mile bugs squashed! (#761)

Fix the orderbook gossip message rebroadcast check that prevents
GossipEngine from rebroadcasting too many messages

Gauthamastro committed:
Merge branch 'Develop' into feature/762/documentation

# Conflicts:
#	clients/orderbook/rpc/src/lib.rs
#	clients/orderbook/src/worker.rs
#	clients/orderbook/src/worker_tests.rs
#	clients/thea/src/lib.rs
#	primitives/thea/src/types.rs

zktony committed:
Update run-a-validator.md

Gauthamastro committed:
Resolve conflicts

Gauthamastro committed:
taplo fmt

Gauthamastro committed:
Extended documentation (#775)

## Describe your changes
Multiple crates, modules and just public API in overall were lacking any
documentation.
In this PR was added documentation on the components listed in the
related issue and added much more to missing parts.

Plus unified format of the docblocks to make them consistent.
## Issue ticket number and link
#762 
## Checklist before requesting a review
- [X] I have performed a self-review of my code.
- [X] If it is a core feature, I have added thorough tests.
- [X] I removed all Clippy and Formatting Warnings. 
- [X] I added required Copyrights.

Gauthamastro committed:
Reorder CI commands

Gauthamastro committed:
Get last finalzied blocks

serhii-temchenko committed:
build(ci): Fixed ordering of build execution

Gauthamastro committed:
Fix compilation

Gauthamastro committed:
Update doc

Gauthamastro committed:
Remove gossip validator field

Gauthamastro committed:
fix clippy

Gauthamastro committed:
Get Finalized Blks in Parachain connector (#778)

serhii-temchenko committed:
build(ci): Build in release mode only on develop branch

serhii-temchenko committed:
build(ci): Syntax fix

Gauthamastro committed:
Fix decoding error

Commit:
sc-network-test and test runtime in test build only

Commit:
no misc build by default

Gauthamastro committed:
CI fixes (#779)

- Fixed ordering of build execution.
- Added `--release` arg to node build.
- CI step name typo fixed.

zktony committed:
Update run-a-validator.md

serhii-temchenko committed:
Merge branch 'Develop' into zktony-update-validator-readme

Commit:
lazy_static + hex_lit

Commit:
taplo fix

Gauthamastro committed:
Iv speeed (#780)

## Describe your changes
* `substrate-test-runtime` and it's client are build only in test config
as dev dependencies;
* `misc/*` stuff is no longer required for workspace build;
* using `hex_lit` for static hex! macro (no proc macro involved);
* using `lazy_static` instead of `Lazy` for initialization of rewards;

## Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.
- [x] I removed all Clippy and Formatting Warnings. 
- [x] I added required Copyrights.

Gauthamastro committed:
Speed

Commit:
const array, no lazy_static

Gauthamastro committed:
const array, no lazy_static (#781)

Gauthamastro committed:
Update weights

Gauthamastro committed:
Update weights

Gauthamastro committed:
Update Pallet rewards structure (#783)

Updates Pallet rewards structure and its weights

Gauthamastro committed:
Update run-a-validator.md (#777)

## Describe your changes
Updated run-a-validator.md.

Gauthamastro committed:
Merge branch 'mainnet-release' into Develop

35359595 committed:
const public and verifier fix

Gauthamastro committed:
const public and verifier fix (#784)
